XMLHttpRequest/send-authentication-cors-setrequestheader-no-cred.htm

```diff
@@ -0,0 +1,62 @@
+<!doctype html>
+<html>
+  <head>
+    <title>XMLHttpRequest: send() - "Basic" authenticated CORS request using setRequestHeader() but not setting withCredentials (expects to succeed)</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="/common/utils.js"></script>
+    <!-- These spec references do not make much sense simply because the spec doesn't say very much about this.. -->
+    <link rel="help" href="https://xhr.spec.whatwg.org/#the-setrequestheader()-method" data-tested-assertations="following::ol[1]/li[6]" />
+    <link rel="help" href="https://xhr.spec.whatwg.org/#the-send()-method" data-tested-assertations="following::code[contains(@title,'http-authorization')]/.." />
+  </head>
+  <body>
+    <div id="log"></div>
+    <script>
+    function doTest(desc, pathsuffic, conditionsFunc, errorFunc) {
+      var test = async_test(desc)
+      test.step(function() {
+        var client = new XMLHttpRequest(),
+            urlstart = location.host + location.pathname.replace(/\/[^\/]*$/, '/'),
+            user = token()
+        client.open("GET", location.protocol + "//www1." + urlstart + "resources/" + pathsuffic, false)
+        client.setRequestHeader("x-user", user)
+        client.setRequestHeader("x-pass", 'pass')
+        client.setRequestHeader("Authorization", "Basic " + btoa(user + ":pass"))
+        client.onerror = test.step_func(errorFunc)
+        client.onreadystatechange = function () {
+          if (client.readyState < 4) {return}
+          test.step(conditionsFunc(client, test, user))
```

Comment:
Did you miss the "return function" part of the condition check functions? I'm open to suggestions as to how this code can become simpler and more obvious, but as far as I can tell your suggested fix would end up running no assertations run at all ;) 

(I suppose your actual suggestion is to remove the closure though.)